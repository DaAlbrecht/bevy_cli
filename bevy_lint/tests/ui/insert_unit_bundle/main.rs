#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_unit_bundle)]

use bevy::prelude::*;
use std::f32::consts::PI;

fn main() {
    App::new().add_systems(Startup, test);
}

fn test(mut commands: Commands) {
    commands.spawn((
        Name::new("Decal"),
        Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
        //~^ ERROR: inserted a `Bundle` containing a unit `()` type
        example(),
        //~^ ERROR: inserted a `Bundle` containing a unit `()` type
        (),
        //~^ ERROR: inserted a `Bundle` containing a unit `()` type
        {
            example();

            Transform::from_translation(Vec3::ONE)
        },
    ));
}

fn example() {
    let foo = 1;
}
