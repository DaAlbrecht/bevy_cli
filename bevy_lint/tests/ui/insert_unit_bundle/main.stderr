warning: unused variable: `foo`
  --> tests/ui/insert_unit_bundle/main.rs:30:9
   |
30 |     let foo = 1;
   |         ^^^ help: if this is intentional, prefix it with an underscore: `_foo`
   |
   = note: `#[warn(unused_variables)]` on by default

error: inserted a `Bundle` containing a unit `()` type
  --> tests/ui/insert_unit_bundle/main.rs:15:9
   |
15 |         Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/insert_unit_bundle/main.rs:3:9
   |
3  | #![deny(bevy::insert_unit_bundle)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^

error: inserted a `Bundle` containing a unit `()` type
  --> tests/ui/insert_unit_bundle/main.rs:17:9
   |
17 |         example(),
   |         ^^^^^^^^^

error: inserted a `Bundle` containing a unit `()` type
  --> tests/ui/insert_unit_bundle/main.rs:19:9
   |
19 |         (),
   |         ^^

error: aborting due to 3 previous errors; 1 warning emitted

